### PR TITLE
SYNF-20 Add support for team deployment

### PR DESCRIPTION
**Pull request by thomasyu888**
- Create Team
- Invite members to team
- [x] Add tests

### CODE REVIEW

File: synapseformation/client.py [hunk 1]
```diff
@@ -59,6 +59,21 @@ def _create_synapse_resources(config: dict, creation_cls: SynapseCreation,
         entity = creation_cls.get_or_create_folder(
             name=config['name'], parentId=parentid
         )
+    elif isinstance(config, dict) and config.get('type') == "Team":
+        team = creation_cls.get_or_create_team(
+            name=config['name'], description=config['description'],
+            canPublicJoin=config['can_public_join']
+        )
+        if config.get("invitations") is not None:
+            # TODO: investigate nested forloop.  This is ugly
```
Comment (mfazza) [inline synapseformation/client.py]: This is still really readable to me.  Perhaps there's an itertools way of doing it, but I'm not aware of it!

*did you mean to remove the #TODO?

**Review by mfazza — APPROVED** [review]
LG 🥇

File: synapseformation/client.py (hunk 1 above)
Comment (thomasyu888) [inline synapseformation/client.py]:
```suggestion
```

Comment (thomasyu888) [inline synapseformation/client.py]:
```suggestion
```

**Pull request merged**